pkg/bootstrapper/bootstrapper_test.go

```diff
@@ -107,3 +113,130 @@ func TestPrepKubeletConfForWindows(t *testing.T) {
 		})
 	}
 }
+
+// TestCloudConfExtraction tests if parseIgnitionFileContents can extract the cloud.conf present in a worker ignition
+// file contents and the resulting file is in the expected format with a set of key value pairs.
+// It also confirms the "--cloud-config" option constructed by WMCB is as expected. Example cloud.conf:
+// {
+//	"cloud": "AzurePublicCloud",
+//	"tenantId": "1234a1b2-a1bc-123a-123a-ab1c2de3afgh",
+//	"aadClientId": "",
+//	"aadClientSecret": "",
+//	"aadClientCertPath": "",
+//	"aadClientCertPassword": "",
+//	"useManagedIdentityExtension": true,
+//	"userAssignedIdentityID": "",
+//	"subscriptionId": "1a123456-12ab-123a-1234-abc1d1ab01c0",
+//	"resourceGroup": "winc-test-rg",
+//	"location": "centralus",
+//	"vnetName": "winc-test-vnet",
+//	"vnetResourceGroup": "winc-test-rg",
+//	"subnetName": "winc-test-node-subnet",
+//	"securityGroupName": "winc-test-node-nsg",
+//	"routeTableName": "winc-test-node-routetable",
+//	"primaryAvailabilitySetName": "",
+//	"vmType": "",
+//	"primaryScaleSetName": "",
+//	"cloudProviderBackoff": true,
+//	"cloudProviderBackoffRetries": 0,
+//	"cloudProviderBackoffExponent": 0,
+//	"cloudProviderBackoffDuration": 6,
+//	"cloudProviderBackoffJitter": 0,
+//	"cloudProviderRateLimit": true,
+//	"cloudProviderRateLimitQPS": 6,
+//	"cloudProviderRateLimitBucket": 10,
+//	"cloudProviderRateLimitQPSWrite": 6,
+//	"cloudProviderRateLimitBucketWrite": 10,
+//	"useInstanceMetadata": true,
+//	"loadBalancerSku": "standard",
+//	"excludeMasterFromStandardLB": null,
+//	"disableOutboundSNAT": null,
+//	"maximumLoadBalancerRuleCount": 0
+//}
+func TestCloudConfExtraction(t *testing.T) {
```

Comment:
If possible, please add a -ve test case, where we don't have a `cloud-config` and the file is not created and that option is not added to kubelet args. Perhaps as a next test.